signing_service/signing_service/utils.py

```diff
@@ -12,3 +16,64 @@ def is_valid_public_key(key):
         return True
     except InvalidKeys:
         return False
+
+
+def make_secret_provider_factory(
+    read_command_line=False,
+    env_variable_name=None,
+    use_file=False,
+    base64_convert=False,
+):
+    def wrapper(**kwargs):
+        return SecretProvider(
+            read_command_line,
+            env_variable_name,
+            use_file,
+            base64_convert,
+            **kwargs
+        )
+    return wrapper
+
+
+class SecretProvider(Action):
+
+    def __init__(
+        self,
+        read_command_line,
+        env_variable_name,
+        use_file,
+        base64_convert,
+        option_strings,
+        dest,
+        required=False,
+        help=None  # pylint: disable=redefined-builtin
+    ):
+        self.read_command_line = read_command_line
+        self.env_variable_name = env_variable_name
+        self.use_file = use_file
+        self.base64_convert = base64_convert
+
+        command_line_arguments = 0 if self.env_variable_name is not None else None
```

Comment:
It can also be none,  so `_count` can be misleading too, I have removed separate variable for it completely and moved the condition to where its result is used directly.